### PR TITLE
Prac 08 code review request

### PR DESCRIPTION
**Pull request by hamzah-opal-n**
Please review this work, @Winminoo4444 - check naming, formatting etc 🐈‍⬛

### CODE REVIEW

File: prac_08/box_layout_demo.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    def handle_greet(self):
+        """Handle pressing greet button."""
+        self.root.ids.output_label.text = f"Hello {self.root.ids.input_name.text}"
+
+    def handle_clear(self):
+        """Handle pressing clear button."""
+        self.root.ids.output_label.text = ""
+        self.root.ids.input_name.text = ""
+
```
Comment (Winminoo4444) [inline prac_08/box_layout_demo.py:23]: There is no handling for cases where input_name or output_label might not exist. A try-except block could help avoid crashes in such cases

File: prac_08/convert_miles_km.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+from kivy.app import App
+from kivy.lang import Builder
+from kivy.core.window import Window
+from kivy.properties import StringProperty
+
+MILES_TO_KILOMETRES_RATE = 1.60934
+
+
+class MilesToKilometresConverter(App):
+    """MilesToKilometresConverter is a Kivy App for converting distances in miles to kilometres."""
```
Comment (Winminoo4444) [inline prac_08/convert_miles_km.py:12]: The docstring for MilesToKilometresConverter is clear and descriptive. Good job!

File: prac_08/convert_miles_km.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        kilometres = miles * MILES_TO_KILOMETRES_RATE
+        self.distance_in_kilometres = str(kilometres)
+
+    def get_valid_miles(self):
+        """Ensure that the value in the miles text input is valid."""
+        try:
+            miles = float(self.root.ids.input_number.text)
+            return miles
+        except ValueError:
+            return 0.0
```
Comment (Winminoo4444) [inline prac_08/convert_miles_km.py:34]: Consider adding a check for negative values if the app should only handle positive distances.

File: prac_08/dynamic_labels.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        self.title = "Dynamic Labels"
+        self.root = Builder.load_file('dynamic_labels.kv')
+        self.create_labels()
+        return self.root
+
+    def create_labels(self):
+        """Create labels from data and add them to the GUI."""
+        for name in self.names:
+            new_label = Label(text=name)
+            self.root.ids.main.add_widget(new_label)
```
Comment (Winminoo4444) [inline prac_08/dynamic_labels.py:28]: Add error handling or validation to ensure self.root.ids.main exists before adding widgets.

File: prac_08/squaring.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        self.root = Builder.load_file('squaring.kv')
+        return self.root
+
+    def handle_calculate(self, value):
+        """Handle calculation (could be button press or other call), output result to label widget."""
+        try:
+            result = float(value) ** 2
+            self.root.ids.output_label.text = str(result)
+        except ValueError:
+            pass
```
Comment (Winminoo4444) [inline prac_08/squaring.py:32]: The except ValueError block silently passes. Add user feedback (e.g., updating the label to show an error message like "Invalid input").

**Review by Winminoo4444 — COMMENTED** [review]
Good work brother!

File: prac_08/box_layout_demo.py (hunk 1 above)
Comment (hamzah-opal-n) [reply prac_08/box_layout_demo.py:23]: True. I didn't add it because the question didn't ask for it.

File: prac_08/convert_miles_km.py (hunk 3 above)
Comment (hamzah-opal-n) [reply prac_08/convert_miles_km.py:34]: Possible, but the question didn't specify it.

File: prac_08/dynamic_labels.py (hunk 4 above)
Comment (hamzah-opal-n) [reply prac_08/dynamic_labels.py:28]: Good point, but the question didn't ask for it.

File: prac_08/squaring.py (hunk 5 above)
Comment (hamzah-opal-n) [reply prac_08/squaring.py:32]: Possible, but the question didn't ask for it. The `except ValueError` block is unmodified from the demo file we were asked to use.

**Pull request merged**